src/components/YourCompany/NameTranslations/NameTranslation.vue

```diff
@@ -0,0 +1,314 @@
+              >
+                <span>Cancel</span>
+              </v-btn>
+            </div>
+          </v-flex>
+        </v-layout>
+      </v-flex>
+    </v-layout>
+  </div>
+</template>
+
+<script lang="ts">
+// Libraries
+import { Component, Vue, Prop, Watch, Emit, Mixins } from 'vue-property-decorator'
+
+// Components
+import { ConfirmDialog } from '@/components/dialogs'
+import { ListNameTranslation, AddNameTranslation } from '.'
+
+// Interfaces
+import { ActionBindingIF, ConfirmDialogType } from '@/interfaces'
+
+// Mixins
+import { CommonMixin } from '@/mixins'
+import { Action, Getter } from 'vuex-class'
+
+@Component({
+  components: {
+    AddNameTranslation,
+    ListNameTranslation,
+    ConfirmDialog
+  }
+})
+export default class NameTranslation extends Mixins(CommonMixin) {
+  // Refs
+  $refs!: {
+    confirmTranslationDialog: ConfirmDialogType
+  }
+
+  @Prop({ default: () => [] })
+  private nameTranslations!: Array<string>
+
+  @Prop({ default: () => [] })
+  private originalNameTranslations!: Array<string>
+
+  // Properties
+  private drafTranslations: Array<string> = []
+  private isEditing: boolean = false
+  private isAddingNameTranslation = false
+  private editingNameTranslation = ''
+  private editIndex = -1
+
+  private get hasNameTranslationChange (): boolean {
+    return this.drafTranslations.length !== this.originalNameTranslations.length ||
+      !this.drafTranslations.every((value, index) => {
+        return this.originalNameTranslations[index] === value
+      })
+  }
+
+  private setNameTranslations (): void {
+    this.emitNameTranslations(this.drafTranslations)
+    this.emitHaveChanges(this.hasNameTranslationChange)
+    this.isEditing = false
+  }
+
+  private resetNameTranslations (): void {
+    this.drafTranslations = [ ...this.originalNameTranslations ]
+    this.emitNameTranslations(this.drafTranslations)
+    this.emitHaveChanges(false)
+    this.isEditing = false
+  }
+
+  private cancelNameTranslationCorrection () {
+    const nameTranslations = this.nameTranslations || []
+    const hasUnsavedData = this.drafTranslations.length !== nameTranslations.length ||
+      !this.drafTranslations.every((value, index) => {
+        return nameTranslations[index] === value
+      })
+    if (hasUnsavedData) {
+      this.confirmUnsavedChanges()
+    } else {
+      this.isEditing = false
+    }
+  }
+
+  private confirmUnsavedChanges () {
+    // open confirmation dialog and wait for response
+    this.$refs.confirmTranslationDialog.open(
+      'Unsaved Changes',
+      'You have unsaved changes in your Name Translations. Do you want to save these changes?',
+      {
+        width: '45rem',
+        persistent: true,
+        yes: 'Save changes',
+        no: null,
+        cancel: 'Cancel without saving'
+      }
+    ).then(async (confirm) => {
+      if (confirm) {
+        this.setNameTranslations()
+      }
+    }).catch(() => {
+      this.onNameTranslationsPropValueChanged()
```

Comment:
Would it make more sense to actually update the prop value here instead of calling the watch handler?